src/frontend/src/flows/addDevice/welcomeView/registerTentativeDevice.ts

```diff
@@ -37,7 +37,7 @@ const displayAlreadyRegisteredDevice = () =>
 export const registerTentativeDevice = async (
   userNumber: bigint,
   connection: Connection
-): Promise<"ok"> => {
+): Promise<{ userNumber: bigint; alias: string } | never> => {
```

Comment:
doesn't this just collapse to `Promise<{ userNumber: bigint; alias: string }>`?

Reply:
Yes but it's more readable. That way I understand straight from the begin that this function might never returns.